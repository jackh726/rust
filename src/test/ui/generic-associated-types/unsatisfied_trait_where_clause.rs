// Ensures that we error for unstatisfied trait where clauses when there isn't
// that clause on the impl.

// check-fail

#![feature(generic_associated_types)]

trait Foo {
    type Assoc<'a, 'b> where 'a: 'b;
}
impl Foo for () {
    type Assoc<'a, 'b> = ();
}

fn make<'a, 'b>() -> <() as Foo>::Assoc::<'a, 'b> {}
//~^ lifetime bound not satisfied

fn main() {}
