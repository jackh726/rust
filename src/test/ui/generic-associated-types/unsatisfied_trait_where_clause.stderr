error[E0478]: lifetime bound not satisfied
  --> $DIR/unsatisfied_trait_where_clause.rs:15:4
   |
LL | fn make<'a, 'b>() -> <() as Foo>::Assoc::<'a, 'b> {}
   |    ^^^^
   |
note: lifetime parameter instantiated with the lifetime `'a` as defined here
  --> $DIR/unsatisfied_trait_where_clause.rs:15:9
   |
LL | fn make<'a, 'b>() -> <() as Foo>::Assoc::<'a, 'b> {}
   |         ^^
note: but lifetime parameter must outlive the lifetime `'b` as defined here
  --> $DIR/unsatisfied_trait_where_clause.rs:15:13
   |
LL | fn make<'a, 'b>() -> <() as Foo>::Assoc::<'a, 'b> {}
   |             ^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0478`.
